File: VB/ConsoleApplication/Model2.vb
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace ConsoleApplication

    'class persisted in DB2
    Public Class Task
        Inherits XPLiteObject

        Private fTaskID As Integer
        <Key(True)> _
        Public Property TaskID() As Integer
            Get
                Return fTaskID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue("TaskID", fTaskID, value)
            End Set
        End Property
        Private fSubject As String
        Public Property Subject() As String
            Get
                Return fSubject
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Subject", fSubject, value)
            End Set
        End Property
        Private fDone As Boolean
        Public Property Done() As Boolean
            Get
                Return fDone
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue("Done", fDone, value)
            End Set
        End Property

        <Browsable(False)> _
        Public ContactID? As Integer

        Public Function GetContact(ByVal foreignSession As Session) As Contact
            Return If(ContactID.HasValue, foreignSession.GetObjectByKey(Of Contact)(ContactID), Nothing)
        End Function
        Public Sub SetContact(ByVal contact As Contact)
            ContactID = If(contact Is Nothing, DirectCast(Nothing, Integer?), CType(contact.ContactID, Integer?))
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
    End Class

End Namespace

File: CS/ConsoleApplication/Model2.cs
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace ConsoleApplication {

    //class persisted in DB2
    public class Task : XPLiteObject {
        int fTaskID;
        [Key(true)]
        public int TaskID {
            get { return fTaskID; }
            set { SetPropertyValue("TaskID", ref fTaskID, value); }
        }
        string fSubject;
        public string Subject {
            get { return fSubject; }
            set { SetPropertyValue("Subject", ref fSubject, value); }
        }
        bool fDone;
        public bool Done {
            get { return fDone; }
            set { SetPropertyValue("Done", ref fDone, value); }
        }

        [Browsable(false)]
        public int? ContactID;

        public Contact GetContact(Session foreignSession) {
            return ContactID.HasValue ? foreignSession.GetObjectByKey<Contact>(ContactID) : null;
        }
        public void SetContact(Contact contact) {
            ContactID = (contact == null) ? (int?)null : (int?)contact.ContactID;
        }

        public Task(Session session) : base(session) { }
    }

}
